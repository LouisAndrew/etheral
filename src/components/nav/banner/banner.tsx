import React from 'react';
import { Link } from '@reach/router';

import { Text } from 'rebass';

export type Props = {
    bannerBgColor: string;
    bannerLink: string;
    bannerText: string;
    bannerTextColor: string;
};

const Banner: React.FC<Props> = ({
    bannerBgColor,
    bannerLink,
    bannerText,
    bannerTextColor,
}) => {
    return (
        <Link to={bannerLink}>
            <Text
                bg={bannerBgColor}
                color={bannerTextColor}
                px={[3]}
                py={[1, 1, 2]}
                textAlign="center"
                fontFamily="body"
                fontWeight="medium"
                role="banner"
                id="banner"
                fontSize={[0, 0, 1]}
            >
                {bannerText}
            </Text>
        </Link>
    );
};

export { Banner };
