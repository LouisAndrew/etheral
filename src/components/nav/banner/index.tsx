import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';
import { Banner as BannerEl, Props } from './banner';

const Banner: React.FC<{}> = () => {
    const data = useStaticQuery(graphql`
        query {
            homepage {
                banner {
                    bannerBgColor
                    bannerLink
                    bannerText
                    bannerTextColor
                }
            }
        }
    `);

    const bannerData = data.homepage.banner as Props;

    return <BannerEl {...bannerData} />;
};

export default Banner;
