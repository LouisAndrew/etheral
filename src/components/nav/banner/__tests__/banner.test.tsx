import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Banner } from '../banner';

describe('Banner', () => {
    const mockBannerAttrs = {
        bannerBgColor: '#fff',
        bannerLink: '/payment',
        bannerText: 'Hello world',
        bannerTextColor: '#000',
    };

    const Element = <Banner {...mockBannerAttrs} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    it('should render the banner with attributes correctly', () => {
        const { queryByRole } = render(Element);
        const BannerElement = queryByRole('banner');

        if (BannerElement) {
            expect(BannerElement).toHaveTextContent(mockBannerAttrs.bannerText);
            expect(BannerElement).toHaveStyle(`
				background-color: ${mockBannerAttrs.bannerBgColor};
				color: ${mockBannerAttrs.bannerTextColor}
			`);
            // expect((BannerElement as HTMLLinkElement).href).toBe(
            //     mockBannerAttrs.bannerLink
            // );
        }
    });

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
