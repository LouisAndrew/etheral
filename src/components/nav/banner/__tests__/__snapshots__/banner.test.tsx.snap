// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches snapshot 1`] = `
<a
  href="/payment"
  onClick={[Function]}
>
  <div
    className="css-ombd65"
    id="banner"
    role="banner"
  >
    Hello world
  </div>
</a>
`;
